src/engine/effects/engineeffectchain.cpp

```diff
@@ -165,98 +165,77 @@ void EngineEffectChain::process(const ChannelHandle& handle,
 
     EffectProcessor::EnableState effectiveEnableState = channel_info.enable_state;
 
+    if (channel_info.enable_state == EffectProcessor::DISABLING) {
+        channel_info.enable_state = EffectProcessor::DISABLED;
```

Comment:
This may be outside the scope of this PR, but temporal effects like echo and reverb should continue to output what is left in their buffers after their enable switch is turned off to have a smooth tail out of the effect. This is really helpful for transitioning between tracks that are much different tempos.